vcd/resource_vcd_nsxt_route_advertisement.go

```diff
@@ -0,0 +1,168 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"log"
+	"strings"
+)
+
+func resourceVcdNsxtRouteAdvertisement() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNsxtRouteAdvertisementCreateUpdate,
+		ReadContext:   resourceVcdNsxtRouteAdvertisementRead,
+		UpdateContext: resourceVcdNsxtRouteAdvertisementCreateUpdate,
+		DeleteContext: resourceVcdNsxtRouteAdvertisementDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNsxtRouteAdvertisementImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+				Deprecated:  "Edge Gateway will be looked up based on 'edge_gateway_id' field",
```

Comment:
Good catch, @vbauzysvmware 

@mikeletux Do I understand right that there is no need for `vdc` field for this resource, because it is traversed from the required EG?

If yes, what is the reason to add this field?

CC: @Didainius

Reply:
Removed VDC as you were right, it's not needed for this resource. 

Take into account that for import is still there, as the URL is org.vdc-or-vdc-group.gateway-name as we don't use IDs for edge GW